.gitignore

```diff
@@ -3,3 +3,5 @@ node_modules
 *.tsbuildinfo
 
 !eslint.config.js
+coverage
+src/generated-types
```

Comment:
I know `typechain` themselves say you shouldn't commit the generated types, but I'm not sure I agree with that. Two reasons:

1. This means we need to carry around `typechain` and `@typechain/ethers-v6` as `dependencies`
2. The output should be deterministic based on ABIs and the specific build dependencies, so unless those two change, we are wasting users' resources

In `subsquid-client` we also track the generated files in the repo, which also makes changes obvious so we can gauge whether an update necessitates major, minor or patch release